packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { AnyFunctionWithRecord, IdempotencyRecordStatus } from './types';
+import { IdempotencyOptions } from './IdempotencyOptions';
+import { IdempotencyRecord } from 'persistence';
+import { IdempotencyInconsistentStateError, IdempotencyItemAlreadyExistsError, IdempotencyAlreadyInProgressError, IdempotencyPersistenceLayerError } from './Exceptions';
+
+export class IdempotencyHandler<U> {
+
+  public constructor(private functiontoMakeIdempotent: AnyFunctionWithRecord<U>, private functionPayloadToBeHashed: unknown, 
+    private idempotencyOptions: IdempotencyOptions, private fullFunctionPayload: Record<string, any>) {}
+
+  public determineResultFromIdempotencyRecord(idempotencyRecord: IdempotencyRecord): Promise<U> | U{ 
```

Comment:
This might be work for another PR, but we need to add comments to the public-facing methods that document use-cases